src/main/java/com/google/sps/data/Card.java

```diff
@@ -107,7 +118,65 @@ public Entity createEntity() {
     card.setProperty("imageBlobKey", this.imageBlobKey);
     card.setProperty("rawText", this.rawText);
     card.setProperty("textTranslated", this.textTranslated);
+    card.setProperty("status", true);
 
     return card;
   }
+
+  public static Card storeCardInDatastore(Card card, DatastoreService datastore, String folderKey) {
+    card.setParentKey(folderKey);
+    Entity cardEntity = card.createEntity();
+    datastore.put(cardEntity);
+
+    card.setCardKey(KeyFactory.keyToString(cardEntity.getKey()));
+
+    return card;
+  }
+
+  public static void deleteCard(Entity card) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    markCardAsDeleted(card);
+    try {
+      datastore.delete(card.getKey());
+    } catch (DatastoreFailureException e) {
+      throw e;
+    }
+  }
+
+  public static void markCardAsDeleted(Entity card) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    card.setProperty("status", false);
+    datastore.put(card);
+  }
+
+  public static void deleteBlob(String blobKey) throws IOException {
```

Comment:
This method doesn't seem like it has anything to do with the Card class.

Reply:
factored out into DeleteFolderServlet.java